src/partisan_hiscamp_membership_strategy.erl

```diff
@@ -0,0 +1,251 @@
+-module(partisan_hiscamp_membership_strategy).
+-author("Hana Frluckaj <hanafrla@cmu.edu>").
+-behavior(partisan_membership_strategy).
+%%import Heap stuff
+-import(heaps, [add/2, delete_min/1, from_list/1, empty/1, merge/2, new/0, min/1,
+                sort/1, to_list/1]).
+%change size to heaps:size()
+% change export list
+%-export([init/1, distance/2, hierarchial_clustering/2]).
+-include("partisan.hrl").
+
+%fix this
+-record(hiScamp, {membership, actor, heap, level1, level2}).
+
+%%initialize state
+init(Identity) -> 
+    Heap = heaps:new(),
+    L1 = sets:new(),
+    L2 = sets:new(),
+    %dataset and data size
+    %gold standard
+    Membership = sets:add_element(myself(), sets:new()),
+    State = #hiScamp{membership=Membership, actor=Identity, heap=Heap, level1 = L1, level2 = L2},
+    MembershipList = membership_list(State),
+    {ok, MembershipList, State}.
+
+%% remote node connects, notify node to us. perform forwarding?
+%% Membership0 is a set of node elements
+join(#hiScamp{membership=Membership0, level1=L1, level2=L2}=State0, Node, _NodeState) ->
+    OutgoingMessages0 = [],
+    case partisan_config:get(tracing, ?TRACING) of
+        true ->
+            lager:info("~p: Adding node ~p to our membership.", [node90, Node]);
+        false ->
+            ok
+    end,
+    Membership = sets:add_element(Node, Membership0),
+%% notify node to add us to its state.
+    Myself = partisan_peer_service_manager:myself(),
+    OutgoingMessages1 = OutgoingMessages0 ++ [{Node, {protocol, {forward_subscription, Myself}}}],
+    OutgoingMessages2 = sets:fold(fun(N, OM) ->
+        %% tracing is a macro
+        case partisan_config:get(tracing, ?TRACING) of
+            true ->
+                lager:info("~p: Forwarding subscription for ~p to node: ~p", [node(), Node, N]);
+            false ->
+                ok
+            end,
+            OM ++ [{N, {protocol, {forward_subscription, Node}}}]
+            end, OutgoingMessages1, Membership0),
+    C = partisan_config:get(scamp_c, ?SCAMP_C_VALUE),
+    ForwardMessages = lists:map(fun(N) ->
+        case partisan_config:get(tracing, ?TRACING) of
+            true ->
+                lager:info("~p: Forwarding additional subs. for ~p to node: ~p", [node(), Node, N]);
+            false ->
+                ok
+        end,
+        {N, {protocol, {forward_subscription, Node}}}
+        end, select_random_sublist(state0, C)),
+    OutgoingMessages = OutgoingMessages2 ++ ForwardMessages,
+    State = State0#hiScamp{membership=Membership},
+    MembershipList = membership_list(State),
+    {ok, Membership, OutgoingMessages, State}.
+
+distance(#hiScamp{membership=Membership0, heap=Heap, level1=L1, level2=L2}=_State0, Node) ->
+    % A node j joins the system by sending a subscription request to the node s which is closest to it
+    Threshold = 3,
+    Dict = get(distance_metrics),
+    Dist = dict:find(Node, Dict),
```

Comment:
```find``` returns ```{ok, Value}``` or ```error```, and you're not handling either case, so L72 will never be true when it should be.